src/git.rs

```diff
@@ -34,7 +37,13 @@ impl GitManagement for Git {
             .and_then(|oid| repo.find_commit(oid.unwrap()))?;
 
         // Create new branch if it doesn't exist
-        repo.branch(branch_name, &commit, false)?;
+        match repo.branch(branch_name, &commit, false) {
+            // This command can fail due to an existing reference. This error should be ignored.
+            Err(err) if err.class() != ErrorClass::Reference || err.code() != ErrorCode::Exists => {
```

Comment:
`||` -> `&&`? We only want to catch this specific error right?

Reply:
Exactly, we only want to catch this error, however this does the opposite, let's the error go through. Following De Morgan's laws this code is equivalent to:

```rust
Err(err) if !(err.class() == ErrorClass::Reference && err.code() == ErrorCode::Exists) => {
```

Maybe that would be a bit more readable, I will change it.